File: day-02-2/java/src/main/java/com/adventofcode/Answer.java
package com.adventofcode;

import java.io.FileInputStream;
import java.io.IOException;

class Answer {

    public static String readInputFile() {
        StringBuilder strBuilder = new StringBuilder();
        int readVar;

        try (FileInputStream fileInput = new FileInputStream("input.txt")) {
            do {
                readVar = fileInput.read();
                if (readVar != -1) {
                    strBuilder.append((char) readVar);
                }
            } while (readVar != -1);
        } catch (IOException e) {
            System.out.println("I/O Error: " + e);
            return "";
        }

        return strBuilder.toString();
    }

    public int solution() {
        String[] commands = readInputFile().split("\n");

        int aim = 0;
        int depth = 0;
        int horizontal = 0;

        for (String command : commands) {
            String[] commandArray = command.split(" ");
            String word = commandArray[0];
            int number = Integer.parseInt(commandArray[1]);

            switch (word) {
                case "down":
                    aim += number;
                    break;

                case "up":
                    aim -= number;
                    break;

                case "forward":
                    horizontal += number;
                    depth += aim * number;
                    break;

                default:
                    break;
            }
        }

        return depth * horizontal;
    }

    public static void main(String[] args) {
        System.out.println(new Answer().solution());
    }

}

File: day-02-2/ruby/answer.rb
# frozen_string_literal: true

class Answer
  def self.read_input_file
    File.read('input.txt')
  end

  def main
    commands = self.class.read_input_file.split("\n")

    aim = depth = horizontal = 0

    commands.each do |command|
      word, number = command.split(' ')
      number = number.to_i

      if word == 'forward'
        horizontal += number
        depth += aim * number
      end

      aim += number if word == 'down'
      aim -= number if word == 'up'
    end

    depth * horizontal
  end
end

puts Answer.new.main
